src/autoIncrement.ts

```diff
@@ -131,10 +144,22 @@ export function AutoIncrementID(schema: mongoose.Schema<any>, options: AutoIncre
         .exec();
 
       if (!counter) {
+        let newCount: bigint = 0n;
+
+        // need to explicitly type check since subtraction is not supported on union number types
+        if (typeof opt.startAt === 'bigint' && typeof opt.incrementBy === 'bigint') {
+          newCount = opt.startAt - opt.incrementBy;
+        } else if (typeof opt.startAt === 'number' && typeof opt.incrementBy === 'number') {
+          newCount = BigInt(opt.startAt - opt.incrementBy);
+        } else {
+          // this should never happen, but checking anyway
+          throw new Error(`Field "${opt.field}" has mismatched "startAt" and "incrementBy" types!`);
+        }
```

Comment:
isnt this already checked in line 107-110?

Reply:
It is, but TypeScript throws the error `Operator '-' cannot be applied to types 'number | bigint' and 'number | bigint'. ts(2365)` unless it is checked here, again. I could omit the final `else` but I didn't think it would do any harm to have it

Reply:
Even with these changes TypeScript still complains with the same error. It seems to be because the `opt.startAt - opt.incrementBy` is happening inside the pre save callback, since if I try to do the same operations outside of the callback in the same scope where `incrementBy` and `startAt` are converted then it doesn't throw this error

Moving the conversion into the pre save callback removes the error though, and all tests pass still, so it doesn't seem like that causes any issues?

Reply:
> Moving the conversion into the pre save callback removes the error though, and all tests pass still, so it doesn't seem like that causes any issues?

it shouldnt cause issues, though i dont think it is really necessary to always convert it to BigInt over and over when once would be enough (though that would seemingly not be enough for typescript), but i think the current way will work for now